// fm-app-settings/src/utils/settings.js
import Cookie from 'js-cookie'
import { getCookie } from './cookie'

const NAMESPACE = 'fm-app-settings'

const ID = () => `_${Math.random().toString(36).substr(2, 9)}`

export class Session {
  constructor(namespace) {
    this.namespace = namespace
  }

  defaultNamespace() {
    // session based unique namespace
    let namespace = Cookie.get(`${ NAMESPACE }.namespace`)
    if (!namespace) {
      namespace = ID()
    }
    return namespace
  }

  get namespace() {
    return this._namespace
  }

  set namespace(namespace) {
    if (!namespace) {
      namespace = this.defaultNamespace()
    }
    this._namespace = namespace
    Cookie.set(`${ NAMESPACE }.namespace`, namespace)
  }

  getMemoryKey(key) {
    return `${ NAMESPACE }.${ this.namespace }.${key}`
  }

  save(key, value) {
    Cookie.set(this.getMemoryKey(key), value)
  }

  memory(key) {
    return Cookie.getJSON(this.getMemoryKey(key))
  }

  forget(key) {
    Cookie.remove(this.getMemoryKey(key))
  }

  getSettingsFromRequest(req) {
    let settings = getCookie(this.getMemoryKey('settings'), true, req.headers)
    if (settings) {
      return settings
    }
  }
}
